### DISCUSSION TITLE
coap will stop working

### DISCUSSION THREAD

**Discussion by aliakseika** [openthread/openthread#9470] [Q&A]
Hello, from time to time something strange happens to the network. Coap messages will stop leaving. In the logs, when the connection is lost, I see only one error: Mac-----------: Frame rx failed, error:UnknownNeighbor. Maybe it's not because of this. I've already broken my head trying to solve the problem. Restarting otbr-agent does not help. It only helps if you turn off the device, twitch the RCP whistle and restart the service. It happens that several messages will break into the network after a freeze, and sometimes not. I am ready to provide any data for analysis, starting with the source code of the firmware (although it works for some time, and only then everything dies, but maybe that’s the problem), and ending with the otbr logs. I would be very grateful for directions on where to look and what to do.
Also, if a network is created, then otbr-agent can restart itself

PS: I apologize for my English, I am writing using Google translator. If something is not clear, then I’m ready to tell you in more detail.

**Comment by aliakseika** [comment]
sometimes such a picture is observed [screen](https://prnt.sc/cTdET4JNMqbk)

**Comment by jwhui** [comment]
Can you share any logs and packet sniffer captures for analysis in Wireshark?

**Comment by aliakseika** [comment]
[logs and code](https://1drv.ms/f/s!AiO65YjMmX8JiuMrJF09poQfCK37BQ?e=7P1Q0e)
Unfortunately, I can't run wireshark because I'm on Macos.

**Reply by jwhui** [reply]
> Unfortunately, I can't run wireshark because I'm on Macos.

Wireshark supports macOS

Can you attach the logs (and packet sniffer capture) directly here?

**Comment by aliakseika** [comment]
yes, but the texas instrument sniffer tools plugin is only for windows. I'll have a Windows computer on Monday, I'll try to configure it. Just don't disappear, I really hope that the problem can be solved

**Comment by aliakseika** [comment]
I installed Wireshark and configured it seemingly as in the guide, but I can’t figure out how to launch it? Which interface should I choose?

**Comment by aliakseika** [comment]
It was not possible to configure wireshark on any system, then one or another error, then the interface does not appear. I had to buy ubiqua for a month, now I’m waiting for the problem to appear again and send the logs.

**Comment by aliakseika** [comment]
[logs](https://1drv.ms/f/s!AiO65YjMmX8JiuMmFlPnD3zioWZa6Q?e=P9Vbku)
The last successful message from coap was at 15:35:51. in the ubiqua log the time is set to UTC, there it is displayed as 12:35:51 for some reason.  The logs are quite large, so I don’t post them in text.

**Reply by jwhui** [reply]
It seems like the device is losing connectivity periodically:
```
$ grep "Link Request" otbr.log
Oct  2 15:06:52 ubuntu otbr-agent[1281]: 02:44:58.604 [I] Mle-----------: Receive Link Request (fe80:0:0:0:2849:41fd:cd18:7079)
Oct  2 15:31:51 ubuntu otbr-agent[1281]: 03:09:57.305 [I] Mle-----------: Receive Link Request (fe80:0:0:0:2849:41fd:cd18:7079)
Oct  2 15:37:01 ubuntu otbr-agent[1281]: 03:15:07.622 [I] Mle-----------: Receive Link Request (fe80:0:0:0:2849:41fd:cd18:7079)
Oct  2 15:38:21 ubuntu otbr-agent[1281]: 03:16:27.831 [I] Mle-----------: Receive Link Request (fe80:0:0:0:2849:41fd:cd18:7079)
Oct  2 15:39:42 ubuntu otbr-agent[1281]: 03:17:48.043 [I] Mle-----------: Receive Link Request (fe80:0:0:0:2849:41fd:cd18:7079)
Oct  2 15:41:02 ubuntu otbr-agent[1281]: 03:19:08.252 [I] Mle-----------: Receive Link Request (fe80:0:0:0:2849:41fd:cd18:7079)
Oct  2 15:42:22 ubuntu otbr-agent[1281]: 03:20:28.459 [I] Mle-----------: Receive Link Request (fe80:0:0:0:2849:41fd:cd18:7079)
Oct  2 15:43:42 ubuntu otbr-agent[1281]: 03:21:48.671 [I] Mle-----------: Receive Link Request (fe80:0:0:0:2849:41fd:cd18:7079)
Oct  2 15:45:02 ubuntu otbr-agent[1281]: 03:23:08.878 [I] Mle-----------: Receive Link Request (fe80:0:0:0:2849:41fd:cd18:7079)
```
This typically occurs when unicast transmissions to a neighboring device are not acknowledged. It is difficult to determine the exact cause without more visibility. In many cases, it is due to low-level radio driver issues.

You could try using a different IEEE 802.15.4 radio platform to see if that helps.

**Comment by aliakseika** [comment]
Do you mean specifically the RCP device, try installing another one?

**Reply by jwhui** [reply]
Both the RCP and the neighboring device.

BTW, you can respond directly to prior comments without suggesting a new answer. Keeping the conversation threaded makes it easier to follow.

**Reply by aliakseika** [reply]
I understand that, unfortunately, I cannot transfer the device itself to another manufacturer; transferring the code from TI to something else is not so easy for me). but I'll try to change the RCP tomorrow to NRF52840
